Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -45,10 +45,7 @@ class SourceGenerator {
         let fileReference = fileReferencesByPath[path]!
         var settings: [String: Any] = [:]
         let buildPhase = buildPhase ?? getDefaultBuildPhase(for: path)
-
-        if buildPhase == .headers {
```

Comment:
This change will require a major version update because you are changing the behavior of the generation. What do you think about keeping it public by default and discuss further if it makes sense to default it to project in a future major version?